http/codegen/server.go

```diff
@@ -511,6 +512,14 @@ func {{ .HandlerInit }}(
 	{{- if .Method.SkipResponseBodyEncodeDecode }}
 		o := res.(*{{ .ServicePkgName }}.{{ .Method.ResponseStruct }})
 		defer o.Body.Close()
+		// handle immediate read error like a returned error
+		buf := &bytes.Buffer{}
+		if _, err = buf.ReadFrom(io.LimitReader(o.Body, 8*1024)); err != nil{
```

Comment:
Should we have a real constant, or a configurable peek value instead of the magic 8*1024?

Reply:
My vote would be a real constant until we see a real need for more configuration which might not be trivial to add in a backwards compatible way...

Reply:
Switching to bufio lets us read 1 byte with impunity. Shouldn't need a constant for 1 :)